pkg/payload/payload.go

```diff
@@ -100,6 +100,11 @@ const (
 	DefaultClusterProfile = "self-managed-high-availability"
 )
 
+type ManifestWithCapabilities struct {
+	Man          manifest.Manifest
+	Capabilities []configv1.ClusterVersionCapability
```

Comment:
Hmm, do we want to defer this until we get openshift/library-go#1333 ?  Because it seems like unnecessary churn to land this shift, then land openshift/library-go#1333 , and then roll back this shift.  But it's not clear to me how invasive this particular pivot is.  If it's fairly small, doing that dance might be worth not blocking on 1333.

Reply:
I didn't want to wait on that. I thought it more important to get CVO working the way we think it should regardless of where some of the code ends up. If I don't do it this way then I'd have to read the manifest annotations again at this point which I know will be dead code and churn anyway.

Reply:
Oh, hmm, yes, we will need something in the manifest library about scraping caps of sets of manifests...

Reply:
ok, thinking this through a bit more, I think openshift/library-go#1333`'s `Include` is sufficient to cover your current `capability.ResourceEnabled` and `ManifestWithCapabilities` functionality.  We'll need something new for update-tainting, to compare the incoming manifest set with the outgoing included manifest set and figure out what caps to taint in, but we don't need that for this PR's non-update `spec.capabilities` case.